Firebase/Auth/Source/FIRUser.m

```diff
@@ -204,6 +209,15 @@ - (nullable instancetype)initWithUser:(FIRUser *)user NS_DESIGNATED_INITIALIZER;
 
 @end
 
+@interface FIRUser ()
+
+/** @property tenantID
+    @brief The tenant ID of the current user. nil if none is available.
+ */
+@property(nonatomic, readwrite, nullable) NSString *tenantID;
```

Comment:
I don't like the idea of having ivars everywhere. Generally ivar should only appear in getter/setter and init, and all other places should use the getter/setter method. The current project has it's legacy and doesn't follow this practice, but it doesn't mean it's a good approach.

In the retrieveUserWithAuth: method the setter method call should be used. using user->_tenantID sounds too old school C style to me.

Reply:
Oh, I also want to add that the current project is a nightmare when executing async blocks because of having ivars everywhere. A good way of dong async blocks is having validity of self pointer act as the condition for execution. So the correct way is to 

WEAKIFY(self);
[block: ^{
  STRONGIFY(self);
  [self.property foo];
}];

but having ivars everywhere makes using them so easy so people tend to write 

WEAKIFY(self);
[block: ^{
  STRONGIFY(self);
  [self.property foo];
  [_ivar bar]; // oops, retain cycle
}];

I have found numerous retain cycles like this in the past three months working on this project during either code review or debugging. If ivars are not abused like this the project will have far fewer retain cycles / execution bugs.